clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -628,7 +631,14 @@ public void setInvalidationValve(InvalidationValve valve) {
       TimeSource timeSource = serviceProvider.getService(TimeSourceService.class).getTimeSource();
 
       OperationsCodec<K, V> codec = new OperationsCodec<>(storeConfig.getKeySerializer(), storeConfig.getValueSerializer());
-      ChainResolver<K, V> resolver = new ChainResolver<>(codec, storeConfig.getExpiry());
+
+      ChainResolver<K, V> resolver;
+      Expiry<? super K, ? super V> expiry = storeConfig.getExpiry();
+      if (expiry == Expirations.noExpiration()) {
```

Comment:
equals in case someone did the no expiration manually instead of using the constant